ci/docker-run.sh

```diff
@@ -78,7 +78,9 @@ ARGS+=(
 # Also propagate environment variables needed for codecov
 # https://docs.codecov.io/docs/testing-with-docker#section-codecov-inside-docker
 # We normalize CI to `1`; but codecov expects it to be `true` to detect Buildkite...
-CODECOV_ENVS=$(CI=true bash <(curl -s https://codecov.io/env))
+# Unfortunately, codecov.io fails sometimes:
+#   curl: (7) Failed to connect to codecov.io port 443: Connection timed out
+CODECOV_ENVS=$(CI=true bash <(while ! curl -sS --retry 5 --retry-delay 2 --retry-connrefused https://codecov.io/env; do sleep 10; done))
```

Comment:
When curl start to provide `--retry-flaky`, the `while` isn't needed.... https://github.com/curl/curl/pull/4558

Reply:
Intended fail-over behavior confirmed: https://buildkite.com/solana-labs/solana/builds/18943#d058e016-9bc2-424c-a48f-e30e626a1b74/70-75

Reply:
Congrats! The upstream `curl` finally added `--retry-all-errors`. Still ubuntu haven't updated it even in the `groovy` release channel, though. We just have to wait more 1.5 years for the next LTS (at least?).

https://curl.haxx.se/changes.html#7_71_0
https://daniel.haxx.se/blog/2020/06/24/curl-7-71-0-blobs-and-retries/
https://curl.haxx.se/docs/manpage.html#--retry-all-errors